### PR TITLE
FYI: Keep to Refer Comments

### PR DESCRIPTION
**Pull request by amit-ib**
Added functionality to edit task

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
     "@types/react": "^18.0.0",
     "@types/react-dom": "^18.0.0",
     "gapi-script": "^1.2.0",
+    "moment": "^2.29.4",
     "node-sass": "^7.0.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-google-login": "^5.2.2",
     "react-icons": "^4.4.0",
+    "react-moment": "^1.1.2",
```
Comment (akash-dhande-ib) [inline package.json:20]: no need of this library, ``moment`` is enough

File: src/assets/scss/components/_list-item.scss [hunk 2]
```diff
@@ -5,10 +5,22 @@
...
+    flex-grow: 1;
+    width: 65%;
+    input {
+      border-radius: 5px;
+      border: solid 1px $gray-color;
+      width: 100%;
+      &[type="date"] {
+        margin-top: 10px;
+      }
+    }
```
Comment (akash-dhande-ib) [inline src/assets/scss/components/_list-item.scss:21]: Please use some generic classes

File: src/components/todo/TodoForm.tsx [hunk 3]
```diff
@@ -13,7 +13,12 @@ const TodoForm = ({ todos, setTodos }: Props) => {
     e.preventDefault();
     if (todotext) {
       const updatedTodos = [...todos];
-      const todo = { id: Date.now(), todo: todotext, isDone: false };
+      const todo = {
+        id: Date.now(),
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoForm.tsx:17]: We have ``moment``, we can use that here.

File: src/components/todo/TodoItem.tsx [hunk 4]
```diff
@@ -16,25 +20,84 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
     setTodos(todos);
   };
 
+  // State to check edit state(if already in edit mode)
+  const [editMode, setEditMode] = useState<boolean>(false);
+  //state to store edit value
+  const [editTodo, setEditTodo] = useState<string>(todoItem.todo);
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:26]: I guess a todo is an object which contains all the data related to that item

File: src/components/todo/TodoItem.tsx [hunk 5]
```diff
@@ -16,25 +20,84 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
+
+  const defaultDate = new Date(editDate).toISOString().split("T")[0]; // yyyy-mm-dd
   return (
     <div className="list-item">
-      <span
-        className={`list-item-text 
+      {editMode ? (
+        //If in edit mode then display Textbox
+        <div className="edit-task-container">
+          <input
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:61]: We can use https://react-hook-form.com/ for forms

File: src/components/todo/TodoItem.tsx [hunk 6]
```diff
@@ -16,25 +20,84 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
+        <span
+          className={`list-item-text 
             ${todoItems[index].isDone ? "task-done" : ""}
           `}
-      >
-        {todoItems[index].todo}
-        <div className="date">13-07-2022</div>
-      </span>
+        >
+          {todoItems[index].todo}
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:81]: redundant code ``todoItems[index]``

File: src/components/todo/TodoItem.tsx [hunk 7]
```diff
@@ -16,25 +20,84 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
       <div className="action-icons">
         <span onClick={() => handleTaskDone(index)}>Mark Done</span>
-        <span className="button-edit">Edit</span>
+        <span
+          className="button-edit"
+          onClick={() => {
+            if (!editMode && !todoItems[index].isDone) {
+              setEditMode(!editMode);
+            }
+          }}
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoItem.tsx:97]: click handler should have function reference only, we should avoid writing whole function here.

**Review by akash-dhande-ib — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (amit-ib) [reply package.json:20]: react-moment - removed

File: src/components/todo/TodoForm.tsx (hunk 3 above)
Comment (amit-ib) [reply src/components/todo/TodoForm.tsx:17]: updated code and used moment

File: src/components/todo/TodoItem.tsx (hunk 6 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: added separate variable for this

File: src/components/todo/TodoItem.tsx (hunk 7 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: called it through a function

File: src/assets/scss/components/_list-item.scss (hunk 2 above)
Comment (amit-ib) [reply src/assets/scss/components/_list-item.scss]: updated

File: src/components/todo/TodoItem.tsx (hunk 4 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: No, todo is just a field name of type string inside TodoModal interface

File: src/components/todo/TodoItem.tsx (hunk 5 above)
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: Updated using react-hook-form

File: package.json (hunk 1 above)
Comment (amit-ib) [reply package.json]: Sorry, my mistake, forgot to update package file

File: src/assets/scss/ui/forms/_button.scss [hunk 8]
```diff
@@ -27,3 +27,10 @@ button {
     padding-right: 10px !important;
   }
 }
+.button-primary {
```
Comment (akash-dhande-ib) [inline src/assets/scss/ui/forms/_button.scss:30]: I guess we already discussed this to maintain it like following

```suggestion
.button{
...
&.primary{
}
&.secondary{
}
}
```

So you won't have redundant code.

**Review by akash-dhande-ib — CHANGES_REQUESTED** [review]

File: src/components/todo/TodoItem.tsx [hunk 9]
```diff
@@ -16,28 +20,97 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
+  const finishEditingTask = (e: any) => {
+    console.log("User pressed: ", e.key);
+    if (e.key === "Enter") {
+      handleTaskEdit(e, todoItem.id, index);
+    }
+  };
+
+  // Enabeling edit mode
+  const enableEditing = () => {
+    if (!editMode && !taskItem.isDone) {
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: yes, you are correct !editMode was not needed, I removed it

File: src/components/todo/TodoItem.tsx [hunk 10]
```diff
@@ -16,28 +20,97 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
           `}
-      >
-        {todoItems[index].todo}
-        <div className="date">13-07-2022</div>
-      </span>
+        >
+          {taskItem.todo}
+
+          <div className="date">
+            {moment(taskItem.date).format("DD/MM/YYYY")}
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: Helper file prepared and used

File: src/components/todo/TodoItem.tsx [hunk 11]
```diff
@@ -16,28 +20,97 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
+          </div>
+        </div>
+      )}
 
       <div className="action-icons">
         <span onClick={() => handleTaskDone(index)}>Mark Done</span>
-        <span className="button-edit">Edit</span>
+        <span
+          className="button-edit"
+          onClick={() => {
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: updated

File: src/components/todo/TodoItem.tsx [hunk 12]
```diff
@@ -16,28 +20,97 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
...
+      todos[index].todo = editTodo;
+      todos[index].date = editDate;
+    }
+    setTodos(todos);
+    setEditMode(false);
+  };
+
+  // function to handel enter key in edit task textbox
+  const finishEditingTask = (e: any) => {
+    console.log("User pressed: ", e.key);
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/components/todo/TodoItem.tsx]: debug removed

File: src/components/todo/TodoForm.tsx (hunk 3 above)
Comment (amit-ib) [reply src/components/todo/TodoForm.tsx]: updated

File: src/components/todo/TodoForm.tsx [hunk 13]
```diff
@@ -13,7 +13,12 @@ const TodoForm = ({ todos, setTodos }: Props) => {
     e.preventDefault();
     if (todotext) {
       const updatedTodos = [...todos];
-      const todo = { id: Date.now(), todo: todotext, isDone: false };
+      const todo = {
+        id: todos.length + 1,
+        todo: todotext,
+        date: new Date(),
```
Comment (akash-dhande-ib) [inline src/components/todo/TodoForm.tsx:19]: ``moment()``

**Comment by akash-dhande-ib** [comment]
Here you would get best opportunity to try resolving conflicts

**Pull request closed without merging**